metrics/time/launch_times.sh

```diff
@@ -38,6 +38,34 @@ REQUIRED_CMDS=("bc" "awk")
 # Note the no-0-padding - 0 padding the results breaks bc in some cases
 DATECMD="date -u +%-s:%-N"
 
+# This the minimum entropy level produced
+# by haveged is 1000 see https://wiki.archlinux.org/index.php/Haveged
+# Less than 1000 could potentially slow down cryptographic
+# applications see https://www.suse.com/support/kb/doc/?id=7011351
```

Comment:
iirc, the real issue we hit for kata boot is the kernel waiting for entropy to be 'enough' before continuing to boot. Let me see if I can dig up some of the details.
The original patch that introduced the 'slow boot' was to fix a CVE - https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=43838a23a05f
This caused boot slowdown hang issues for a number of situations. A follow up patch added RDRAND support to help get around this:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a26fb01c2879ed7026e6cbd78bb701912d249eef
but I think that patch landed in v4.18, and we are not quite on that yet... so we still see the stall/hangs, and hence recommend adding an entropy daemon on the host.
There is an overview here https://www.phoronix.com/scan.php?page=news_item&px=Linux-Protect-User-Entropy

The bit that I think is hard to work out is, how much entropy needs to be available for the boot to actually happen. It's not obvious to me from the code (welcome to the world of crypto/random code... maybe you are not meant to be able to understand it ;-) ).

The '1000' might be enough - @GabyCT , did you test this? Did you try running the test on a machine without the entropy daemon or this patch, and get a 'spiky' result, and then re-run it with just this patch to see if it fixes the issue (just this patch, as if you add in the entropy daemon I think you will always have >1000 and thus always just boot anyway).

Reply:
@grahamwhaley thanks for the explanation, I ran without and with entropy running 1000 containers, I will examine the results running the report generation (for both cases I ran twice). I am also trying to see if 1000 is enough or we need more.

Reply:
Any update on this @GabyCT?